hayeon/programmers-1/숫자-짝꿍.js

```diff
@@ -0,0 +1,38 @@
+function solution(X, Y) {
+  let result = "";
+
+  X = X.split("");
+  Y = Y.split("");
+
+  for (let i = 0; i <= 9; i++) {
+    const numX = X.filter((el) => Number(el) === i).length;
+    const numY = Y.filter((el) => Number(el) === i).length;
+
+    result += String(i).repeat(Math.min(numX, numY));
+  }
+
+  if (result === "") return "-1";
+  if (Number(result) === 0) return "0";
+
+  return result
+    .split("")
+    .sort((a, b) => b - a)
+    .join("");
+}
```

Comment:
```suggestion
function solution(X, Y) {
  const arrX = X.split("");
  const arrY = Y.split("");
  let result = "";

  for (let i = 0; i <= 9; i++) {
    const numX = arrX.filter((el) => Number(el) === i).length;
    const numY = arrY.filter((el) => Number(el) === i).length;

    result += String(i).repeat(Math.min(numX, numY));
  }

  if (result === "") return "-1";
  if (Number(result) === 0) return "0";

  return result
    .split("")
    .sort((a, b) => b - a)
    .join("");
}
```
- 오우 너무 좋은 풀이 잘봤습니다 0~9를 순회하면서 이를 찾아내는 방법도 있었군요 훌륭합니다.
- 다만 최근에 계속 언급했지만 매개변수를 직접 건드리는 것은 웬만하면 지양합시다 👍
  - 현재 풀이에서는 큰 문제가 없지만 좋은 습관을 들여봐요 ㅎㅎ 
- 그리고 좀 아쉬운 것은 for문에서 numX와 numY가 뭐랄까 처음 딱 봤을 때 뭘 의미하는 건지 와닿지가 않네요  🥲 좀더 명확한 변수명은 어떤게 있을까요? 한번 고민해보시고 답변 남겨주세요 😄
- 그리고 `arrX.filter((el) => Number(el) === i).length;` 이 부분도 반복되는 코드라 함수로 분리하면 더 좋은 코드가 될 것 같아요 
- 그리고 윤우님 코드도 참고해주시면 또 다른 접근 방식의 풀이도 확인하실 수 있습니다. 🙇‍♂️

Reply:
저도 변수명이..너무 고민이었습니다....ㅠㅠ xLength를 하려다가 numX로 바꿨는데 뭐가 더 좋을지는 잘 모르겠네유.. 매개변수를 다시 변수로 지정하는 습관은 꼭 들이겠습니다 ㅜㅜ 저도 코드를 반복적으로 써서 뭔가 찜찜했는데 다시 한번 더 생각해보겠습니다 ㅎㅎ 감사합니다!!😆 윤우님 코드도 참고해서 다시 풀어보겠습니닷!